KC7-Foundation/kc7_scoreboard
# Manage users functionality

### Description
This pull request introduces a new feature that enables session managers to add and remove session users from the _Manage Challenge_ page. With this update, session managers will be able to add users in advance, simplifying the process of managing challenge participants.

### Changes made
To implement this feature, the following changes were made:

- Added a dropdown input to enable session managers to add users
- Added a section to display session users
- Created a new route for adding registrants
- Created a new route for removing registrants
- Updated the manage_challenge route so that it now includes session users and all user usernames, allowing session managers to easily add new participants.

### How to test
1. Switch to the manage_users_functionality branch of the repository.
2. Navigate to the _Manage Challenges_ page for a session.
3. Attempt to add and remove users using the newly implemented feature.

Please note that admins will not appear in the registrant's section. Additionally, if a user is already part of the session, they will not appear in the dropdown. Additionally, when you delete an admin, their name will move to the registrant's category as they are still a registrant, but not an admin. To completely remove someone from a game, they but be removed as a registrant.

### Screenshots
| No registrants added | Added registrants |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/97637517/233904276-4c8ec8d4-d445-4cec-a141-88c0df59e30e.png) | ![image](https://user-images.githubusercontent.com/97637517/233904367-a9550553-0e38-4686-b740-87bce149c27b.png) |


Resolves Issue #14
